client/clientutil/modelinfo.go

```diff
@@ -0,0 +1,580 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package clientutil
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+	"text/tabwriter"
+	"time"
+
+	"github.com/snapcore/snapd/asserts"
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/strutil"
+	"github.com/snapcore/snapd/timeutil"
+)
+
+var (
+	invalidTypeMessage = i18n.G("invalid type for %q header")
+	errNoMainAssertion = errors.New(i18n.G("device not ready yet (no assertions found)"))
+	errNoSerial        = errors.New(i18n.G("device not registered yet (no serial assertion found)"))
+
+	// this list is a "nice" "human" "readable" "ordering" of headers to print
+	// off, sorted in lexical order with meta headers and primary key headers
+	// removed, and big nasty keys such as device-key-sha3-384 and
+	// device-key at the bottom
+	// it also contains both serial and model assertion headers, but we
+	// follow the same code path for both assertion types and some of the
+	// headers are shared between the two, so it still works out correctly
+	niceOrdering = [...]string{
+		"architecture",
+		"base",
+		"classic",
+		"display-name",
+		"gadget",
+		"kernel",
+		"revision",
+		"store",
+		"system-user-authority",
+		"timestamp",
+		"required-snaps", // for uc16 and uc18 models
+		"snaps",          // for uc20 models
+		"device-key-sha3-384",
+		"device-key",
+	}
+)
+
+type ModelFormatter interface {
+	GetPublisher() string
+	GetEscapedDash() string
+}
```

Comment:
Do we need this, or is it a carry over from the initial approach? escaped dash could be a const, and publisher can be an input arg for the print methods?

Reply:
this interface is awkward, GetPublisher should become LongPublisher(storeAccountID string)

Reply:
I can see what you guys mean, however GetEscapedDash uses a different implementation (colors.go) from cmd/snap, and I don't want to move colors.go and the code it uses, as it's a bit convoluted. 

GetEscapedDash
When used from cmd/snap, we want it to correctly select the unicode character when --unicode is provided to the command through the x.GetEscapes object from the cmd/snap package.
When used from snapctl, we don't have this switch (can be added tho), so we assume to use the non-unicode character in all scenarios (--), and we don't want all the code from colors.go in that case

GetPublisher
I'll rename this and add the parameter